src/components/pages/devices/flyouts/deviceNew/deviceNew.js

```diff
@@ -401,125 +401,132 @@ export class DeviceNew extends LinkedComponent {
     const summaryMessage = this.getSummaryMessage();
 
     return (
-      <Flyout header={t('devices.flyouts.new.title')} onClose={() => this.onFlyoutClose('Devices_TopXCloseClick')}>
-          <Protected permission={permissions.createDevices}>
-            <form className="devices-new-container" onSubmit={this.apply}>
-              <div className="devices-new-content">
+      <Flyout header={t('devices.flyouts.new.title')} t={t} onClose={() => this.onFlyoutClose('Devices_TopXCloseClick')}>
+        <Protected permission={permissions.createDevices}>
+          <form className="devices-new-container" onSubmit={this.apply}>
+            <div className="devices-new-content">
+              <FormGroup>
+                <FormLabel>{t(deviceOptions.labelName)}</FormLabel>
+                <Radio link={this.deviceLink} value={deviceOptions.device.value} onChange={this.deviceChange}>
+                  {t(deviceOptions.device.labelName)}
+                </Radio>
+                <Radio link={this.deviceLink} value={deviceOptions.edgeDevice.value} onChange={this.deviceChange}>
+                  {t(deviceOptions.edgeDevice.labelName)}
+                </Radio>
+              </FormGroup>
+              {
+                isEdgeDevice &&
                 <FormGroup>
-                  <FormLabel>{t(deviceOptions.labelName)}</FormLabel>
-                  <Radio link={this.deviceLink} value={deviceOptions.device.value} onChange={this.deviceChange}>
-                    {t(deviceOptions.device.labelName)}
+                  <FormLabel>{t(deviceTypeOptions.labelName)}</FormLabel>
+                  <Radio link={this.deviceTypeLink} value={deviceTypeOptions.simulated.value} onChange={this.deviceTypeChange}>
+                    {t(deviceTypeOptions.simulated.labelName)}
                   </Radio>
-                  <Radio link={this.deviceLink} value={deviceOptions.edgeDevice.value} onChange={this.deviceChange}>
-                    {t(deviceOptions.edgeDevice.labelName)}
+                  <Radio link={this.deviceTypeLink} value={deviceTypeOptions.physical.value} onChange={this.deviceTypeChange}>
+                    {t(deviceTypeOptions.physical.labelName)}
                   </Radio>
                 </FormGroup>
-                {
-                  isEdgeDevice &&
+              }
+              {
+                isSimulatedDevice &&
+                <ComponentArray>
```

Comment:
Since the repo has been updated to React 16 you can use react fragments now instead of this component. 

```jsx
<>
 // ...
</>
```

Reply:
This made me think if I ever intentionally updated this file and apparently this is a stale commit made in my master branch months ago. 😈